smartmeter/smartmeter.ino

```diff
@@ -90,6 +90,26 @@ void setupHW()
     digitalWrite(RELAY, OUTPUT);
 }
 
+void screenChange()
+{
+    static uint id = 0;
+    static uint slt = 0;
+    id++;
+    slt = id%(20*5*3); // Time for all screens to share: fps * screen_time * num_screens
+    if(slt<=(20*5*1)) // Time for each screen: fps * screen_time
+    {
+        Display::self().displayLogo();
+    }
+    else if(slt<=(20*5*2))
+    {
+        Display::self().displayData();
+    }
+    else
```

Comment:
A switch case or a vector interaction would be better than hardcoded values.

Reply:
Should we use a fixed value for FPS or should we take the actual FPS calculated inside the displayData function?

Reply:
se set the fps in the constructor with the delay time.

Reply:
Ok. What do you mean by display value? Which line?